### PR TITLE
Fixed bug around channel file sidebar

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Fixes https://mattermost.atlassian.net/browse/MM-58545

Associated enterprise PR - https://github.com/mattermost/enterprise/pull/1705

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-58545

#### Release Note
```release-note
None
```

### CODE REVIEW

**Comment by harshilsharma63** [comment]
Test case are in-progress, but wanted to get a head start with code review.

File: server/platform/services/searchengine/bleveengine/indexer/indexing_job.go [hunk 1]
```diff
@@ -472,7 +472,9 @@ func (worker *BleveIndexerWorker) BulkIndexFiles(files []*model.FileForIndexing,
 	batch := worker.engine.FileIndex.NewBatch()
 
 	for _, file := range files {
-		if file.DeleteAt == 0 {
+		shouldIndexFile := file.DeleteAt == 0 && (file.PostId != "" || file.CreatorId == model.BookmarkFileOwner)
```
Comment (enzowritescode) [inline server/platform/services/searchengine/bleveengine/indexer/indexing_job.go:475]: Pending my comment in the other related PR

**Review by larkox — APPROVED** [review]

File: server/platform/services/searchengine/bleveengine/indexer/indexing_job.go (hunk 1 above)
Comment (harshilsharma63) [reply server/platform/services/searchengine/bleveengine/indexer/indexing_job.go]: Fixed.

**Review by enzowritescode — APPROVED** [review]

**Pull request merged**